clients/vscode/src/Config.ts

```diff
@@ -1,13 +1,23 @@
 import { EventEmitter } from "events";
 import { workspace, ExtensionContext, WorkspaceConfiguration, ConfigurationTarget, Memento } from "vscode";
 import { ClientProvidedConfig } from "tabby-agent";
+import { ContextVariables } from "./ContextVariables";
+
+interface AdvancedSettings {
+  "inlineCompletion.triggerMode"?: "automatic" | "manual";
+}
 
 export class Config extends EventEmitter {
-  constructor(private readonly context: ExtensionContext) {
+  constructor(
+    private readonly context: ExtensionContext,
+    private contextVariables: ContextVariables,
+  ) {
```

Comment:
Don't add ContextVariables as a dependency of Config. Instead, let's add Config to the ContextVariables constructor, initialize the value there, and add a `config.on('update')` listener for updating it.

Reply:
Updated, plz review again